src/backend/marsha/core/api/video.py

```diff
@@ -406,53 +419,42 @@ def end_live(self, request, pk=None):
         video = self.get_object()
 
         if video.live_state is None:
-            return Response({"error": "Impossible to stop a non live video."}, 400)
+            return Response({"error": "Impossible to harvest a non live video."}, 400)
 
-        if video.live_state not in [defaults.IDLE, defaults.PAUSED]:
+        if video.live_state != defaults.STOPPED:
             return Response(
                 {
                     "error": (
-                        "Live video must be stopped before deleting it."
+                        "Live video must be stopped before harvesting it."
                         f" Current status is {video.live_state}"
                     )
                 },
                 400,
             )
 
+        if not video.recording_slices:
+            return Response(
+                {"error": "Live video must be recording before harvesting it."}, 400
+            )
+
         if video.is_recording:
             stop_recording(video)
 
-        if video.live_state == defaults.IDLE:
-            video.upload_state = defaults.DELETED
-            video.live_state = None
-            video.live_info = None
-            video.live_type = None
-            video.save()
-            channel_layers_utils.dispatch_video_to_groups(video)
-            serializer = self.get_serializer(video)
-
-            return Response(serializer.data)
-
         delete_aws_element_stack(video)
         if settings.LIVE_CHAT_ENABLED:
             close_room(video.id)
 
-        if video.allow_recording and video.recording_slices:
-            try:
-                create_mediapackage_harvest_job(video)
-                video.live_state = defaults.STOPPED
-                video.upload_state = defaults.HARVESTING
-            except ManifestMissingException:
-                delete_mediapackage_channel(video.get_mediapackage_channel().get("id"))
-                video.upload_state = defaults.DELETED
-                video.live_state = None
-                video.live_info = None
-                video.live_type = None
-        else:
-            video.delete()
-            video.upload_state = defaults.DELETED
+        try:
+            create_mediapackage_harvest_job(video)
+            video.live_state = defaults.HARVESTING
+        except ManifestMissingException:
+            delete_mediapackage_channel(video.get_mediapackage_channel().get("id"))
+            video.upload_state = defaults.PENDING
             video.live_state = None
```

Comment:
This case should not happen. The `live_state` should be `stopped` here IMO

Reply:
Also having the `ManifestMissingException` should never happen anymore IMO. To start recording we must start the live. Once the live started, the manifest will be created (modulo the latency). Keeping `stopped` should be good enough